### PR TITLE
Add support for enabling KVM HyperV emulation

### PR DESCRIPTION
**Pull request by rbradford**
This when combined with #1701 enables Windows to boot.

### CODE REVIEW

**Comment by liuw** [comment]
There is only one global toggle in this series. What's your thought on picking and choosing individual features? Is it allowed to change the command line options CH supports in a non-backward-compatible way?

**Comment by rbradford** [comment]
> There is only one global toggle in this series. What's your thought on picking and choosing individual features? Is it allowed to change the command line options CH supports in a non-backward-compatible way?

We can add features to this global toggle but I see no value in subdividing it into all the individual knobs that QEMU offers.

**Comment by sameo** [comment]
> There is only one global toggle in this series. What's your thought on picking and choosing individual features? Is it allowed to change the command line options CH supports in a non-backward-compatible way?

I'd rather only have one switch as well. If in the future we clearly identify a need for splitting it into more granular features, we can change that.

File: arch/src/x86_64/mod.rs [hunk 1]
```diff
@@ -333,13 +336,71 @@ pub fn configure_vcpu(
...
     CpuidPatch::set_cpuid_reg(&mut cpuid, 0xb, None, CpuidReg::EDX, u32::from(id));
     CpuidPatch::set_cpuid_reg(&mut cpuid, 0x1f, None, CpuidReg::EDX, u32::from(id));
+
+    if kvm_hyperv {
+        // Remove conflicting entries
+        cpuid.retain(|c| c.function != 0x4000_0000);
+        cpuid.retain(|c| c.function != 0x4000_0001);
+
+        // See "Hypervisor Top Level Functional Specification" for details
+        // Compliance with "Hv#" requires leaves upto 0x4000_000a
```
Comment (liuw) [inline arch/src/x86_64/mod.rs:351]: "upto" should be "up to" and "Hv#" should be "Hv#1".

**Review by liuw — APPROVED** [review]

**Comment by liuw** [comment]
@rbradford @sameo 

Thanks for the clarification. I'm fine with this global toggle approach for now.

There are some other emulation features to improve Windows' performance. I don't see a reason to not enable them. But that can wait till another day.

**Pull request merged**